File: 12936.py
def solution(n, k):
    answer = []

    factorial = [1 for i in range(21)]  # 팩토리얼 구하기
    for i in range(1, 21):
        factorial[i] = factorial[i - 1] * i

    numbers = [i for i in range(1, n + 1)]

    for i in range(n, 0, -1):
        digit = (k - 1) // factorial[i - 1]  # 들어갈 숫자 구하기

        answer.append(numbers[digit])
        numbers.pop(digit)

        k = k % factorial[i - 1]

    return answer

File: 12936.java
import java.util.ArrayList;

class Solution {

    public int[] solution(int n, long k) {
        int index = 0;
        int[] answer = new int[n];
        ArrayList<Integer> numbers = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            numbers.add(i + 1);
        }

        k -= 1;  // 0-index

        while (n > 0) {
            long mod = 1;
            for (int i = 1; i <= n - 1; i++) {
                mod *= i;
            }

            answer[index++] = numbers.remove((int) (k / mod));

            n -= 1;
            k %= mod;
        }

        return answer;
    }
}
